packages/subproviders/src/subproviders/base_wallet_subprovider.ts

```diff
@@ -0,0 +1,134 @@
+import { JSONRPCRequestPayload, JSONRPCResponsePayload } from '@0xproject/types';
+import { addressUtils } from '@0xproject/utils';
+import * as _ from 'lodash';
+
+import { Callback, PartialTxParams, ResponseWithTxParams, WalletSubproviderErrors } from '../types';
+
+import { Subprovider } from './subprovider';
+
+export abstract class BaseWalletSubprovider extends Subprovider {
+    protected static _validateSender(sender: string) {
+        if (_.isUndefined(sender) || !addressUtils.isAddress(sender)) {
+            throw new Error(WalletSubproviderErrors.SenderInvalidOrNotSupplied);
+        }
+    }
+
+    public abstract async getAccountsAsync(): Promise<string[]>;
+    public abstract async signTransactionAsync(txParams: PartialTxParams): Promise<string>;
+    public abstract async signPersonalMessageAsync(dataIfExists: string): Promise<string>;
```

Comment:
the param name `dataIfExists` implies the type is `string | undefined`

Reply:
Will change to just `data` as an undefined value is an error and cannot be handled (but we should still assert it's defined in the implementation)